### ISSUE TITLE
Datagrid: Date picker for cannot select range date with customized filter conditions.

### ISSUE THREAD

**Issue by manonshen** [infor-design/enterprise#8173]
**Describe the bug**
I have a datagrid column with date type, customized filter conditions as below.
The first filter condition when opening datepicker should be range, but it looks like it's using equals.
```
_columns.push({
            id: 'uploadTime',
            name: this.translate.instant('gm.datafilemgr.col.uploadtime'),
            field: DataFile.uploadTime,
            filterType: 'date',
            editorOptions: { showTime: true, useCurrentTime: true, minuteInterval: 1, secondInterval: 1 },
            dateFormat: Soho.Locale.calendar().dateFormat.datetime,
            **filterConditions: ['in-range', 'less-than', 'less-equals', 'greater-than', 'greater-equals'],**
            width: 240,
            sortable: true,
            formatter: TimestampFormatter,
        });_
```
**To Reproduce**

1. In the html(https://main-enterprise.demo.design.infor.com/components/datagrid/example-custom-filter-conditions.html), change the column settings of Date column to follow exactly the filterConditions mentioned above.

2. The first filter condition when opening datepicker should be range, but it looks like it's using equals

**Expected behavior**
The default filter condition is In-range, opening datepicker should be range.

**Version**

- ids-enterprise: [e.g. v4.9.0 or v4.10.0]

**Comment by tjamesallen15** [comment]
Hi @manonshen

I think this was fixed with my last PR in 4.91. -> https://github.com/infor-design/enterprise/issues/8147

Tried porting your settings and it looks fine now.

cc @ericangeles @tmcconechy

**Comment by glenlieorillo** [comment]
This is now working as expected.


https://github.com/infor-design/enterprise/assets/102151352/bf1450a2-7417-45d3-bd21-2b30bb4a575e
